### PR TITLE
windmill: 1.160.0 -> 1.184.0

### PR DESCRIPTION
**Pull request by happysalada**
## Description of changes

update of windmill to latest

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 261813` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>windmill</li>
  </ul>
</details>

**Comment by happysalada** [comment]
marking as draft as this fails with
```
l-ui> 11: 0xe78e2d v8::internal::Factory::NewFillerObject(int, v8::internal::Allo
windmill-ui> 12: 0x12611cb v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned
windmill-ui> 13: 0x16ab8f9  [/nix/store/x8vrc1c6r8nn454whr99bxc5hdy5ybjf-nodejs-18.18.0/
windmill-ui> /nix/store/04skzfj94zkh62h2c592nivc7vj0ld84-npm-build-hook/nix-support/setu
p-hook: line 3:  1375 Aborted                 (core dumped) npm run ${npmWorkspace+--wor
kspace=$npmWorkspace} "$npmBuildScript" $npmBuildFlags "${npmBuildFlagsArray[@]}" $npmFl
ags "${npmFlagsArray[@]}"
windmill-ui>
windmill-ui> ERROR: `npm build` failed
windmill-ui>
windmill-ui> Here are a few things you can try, depending on the error:
windmill-ui> 1. Make sure your build script (build) exists
windmill-ui>   If there is none, set `dontNpmBuild = true`.
windmill-ui> 2. If the error being thrown is something similar to "error:0308010C:digita
l envelope routines::unsupported", add `NODE_OPTIONS = "--openssl-legacy-provider"` to y
our derivation
windmill-ui>   See https://github.com/webpack/webpack/issues/14532 for more information.
windmill-ui>
error: builder for '/nix/store/m2y8xa299l3ccxnzks0sgh7wbfrck1vr-windmill-ui-1.184.0.drv'
 failed with exit code 1;
       last 25 log lines:
       >
       > FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - Java
Script heap out of memory
       >  1: 0xab4728 node::Abort() [/nix/store/x8vrc1c6r8nn454whr99bxc5hdy5ybjf-nodejs-
18.18.0/bin/node]
       >  2: 0x9a4839  [/nix/store/x8vrc1c6r8nn454whr99bxc5hdy5ybjf-nodejs-18.18.0/bin/n
ode]
       >  3: 0xcc1ec0 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, b
ool) [/nix/store/x8vrc1c6r8nn454whr99bxc5hdy5ybjf-nodejs-18.18.0/bin/node]
       >  4: 0xcc228b v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*,
       >  5: 0xea67a5  [/nix/store/x8vrc1c6r8nn454whr99bxc5hdy5ybjf-nodejs-18.18.0/bin/n
       >  6: 0xea687b  [/nix/store/x8vrc1c6r8nn454whr99bxc5hdy5ybjf-nodejs-18.18.0/bin/n
       >  7: 0xeb9c6a v8::internal::Heap::PerformGarbageCollection(v8::internal::Garbage
       >  8: 0xebaa85 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace,
       >  9: 0xe9737a v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath(int
       > 10: 0xe98724 v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath(in
       > 11: 0xe78e2d v8::internal::Factory::NewFillerObject(int, v8::internal::Allocati
       > 12: 0x12611cb v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned lon
       > 13: 0x16ab8f9  [/nix/store/x8vrc1c6r8nn454whr99bxc5hdy5ybjf-nodejs-18.18.0/bin/
       > /nix/store/04skzfj94zkh62h2c592nivc7vj0ld84-npm-build-hook/nix-support/setup-ho
       >
       > ERROR: `npm build` failed
       >
       > Here are a few things you can try, depending on the error:
       > 1. Make sure your build script (build) exists
       >   If there is none, set `dontNpmBuild = true`.
       > 2. If the error being thrown is something similar to "error:0308010C:digital en
       >   See https://github.com/webpack/webpack/issues/14532 for more information.
       >
       For full logs, run 'nix log /nix/store/m2y8xa299l3ccxnzks0sgh7wbfrck1vr-windmill-
error: 1 dependencies of derivation '/nix/store/7s66hw4rgd3h1ipw8w8b89c1kd9ncxwx-windmil
l-1.184.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/byg7419ka6fzf6ydkg2fkv7bb67i2vpi-review-
shell.drv' failed to build
┏━ 1 Errors:
 ⋮
┃        >   If there is none, set `dontNpmBuild = true`.
┃        > 2. If the error being thrown is something similar to "error:0308010C:digital…
┃        >   See https://github.com/webpack/webpack/issues/14532 for more information.
┃        >
┃        For full logs, run 'nix log /nix/store/m2y8xa299l3ccxnzks0sgh7wbfrck1vr-windmi…
┣━ Dependency Graph:
```

I'll try updating again later on. I've never seen this js error.

**Comment by happysalada** [comment]
@dit7ya tagged you just in case you've encountered that error before, but if you haven't, feel free to skip.

**Comment by happysalada** [comment]
@dit7ya I've managed to get it working.
I've also added myself as a maintainer if you don't mind.
let me know if anything.

File: pkgs/servers/windmill/default.nix [hunk 1]
```diff
@@ -43,7 +43,11 @@ let
 
     sourceRoot = "${fullSrc.name}/frontend";
 
-    npmDepsHash = "sha256-GUrOfN3SyxkvQllgHXDao8JFl5zY4DBxftelsX0Rkqo=";
+    npmDepsHash = "sha256-pGTJfVXo7nPIzwVIVxOm1pTd+7CbnKCnaQMYC+GkSAI=";
+
+    # without these you get a
+    # FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
+    NODE_OPTIONS="--max-old-space-size=8192";
```
Comment (dit7ya) [inline pkgs/servers/windmill/default.nix:50]:
```suggestion
env = {
    NODE_OPTIONS="--max-old-space-size=8192";
    }
```

**Review by dit7ya — COMMENTED** [review]
I am taking a little break from maintaining nixpkgs. I added one small comment but haven't tested the package. Feel free to add yourself as a maintainer. Thank you!

File: pkgs/servers/windmill/default.nix (hunk 1 above)
Comment (happysalada) [reply pkgs/servers/windmill/default.nix]: done, thank you for the suggestion.

**Comment by happysalada** [comment]
@dit7ya thanks a lot for the effort that went into packaging this.
Enjoy some well deserved rest.
let me know if there is anything else on this, I'll keep windmill warm until you come back.

**Pull request merged**